packages/react-components/react-badge/library/src/components/Badge/useBadgeStyles.styles.ts

```diff
@@ -321,7 +321,7 @@ export const useBadgeStyles_unstable = (state: BadgeState): BadgeState => {
   const iconStyles = useIconStyles();
   if (state.icon) {
     let iconPositionClass;
-    if (state.root.children) {
+    if (state.root.children !== null && state.root.children !== undefined && state.root.children !== false) {
```

Comment:
the `!= null` check should be enough

```suggestion
    // eslint-disable-next-line eqeqeq
    if (state.root.children != null) {
```